shopfloor/services/delivery.py

```diff
@@ -211,25 +212,24 @@ def _set_lines_done(self, lines, product_qty=None):
         if product_qty:  # defined with lot/product/packaging scan
             # With a product_qty we process only one move line,
             # so one move to deal with regarding the qty
-            qty_done = lines.move_id.product_id.uom_id._compute_quantity(
+            qty_picked = lines.move_id.product_id.uom_id._compute_quantity(
                 product_qty, lines.move_id.product_uom
             )
-            lines.qty_done += qty_done
+            for line in lines:
+                line._pick_qty(line.qty_picked + qty_picked)
             return self._action_picking_done(
                 lines.picking_id, force=allow_prepackaged_product
             )
         for line in lines:
             # note: the package level is automatically set to "is_done" when
             # the qty_done is full
-            line.qty_done = line.quantity
+            line._pick_qty(line.quantity)
```

Comment:
Here we are picking all qty (qty_done = quantity):
```suggestion
            line.picked = True
```